client/castApp/src/castinit.c

```diff
@@ -65,13 +67,124 @@ static void casthook(initHookState state)
+        /* shutdown in progress, silent no-op */
+        epicsMutexUnlock(self->lock);
+        return;
+    }
+    else if(self->current != casterStateInit) {
+        /* Attempt to add after iocInit(), when we may be connected.
+           To fully support, would need to force reconnect or resend w/ updated envs list. */
+        errlogSevPrintf(errlogMinor, "addReccasterEnvVars called after iocInit() when reccaster might already be connected. Not supported\n");
+        epicsMutexUnlock(self->lock);
+        return;
+    }
+    int new_extra_envs_size = self->num_extra_envs + argc;
+    int num_new_extra_envs = self->num_extra_envs;
+
+    char **new_extra_envs = calloc(new_extra_envs_size, sizeof(*new_extra_envs));
+    if(new_extra_envs == NULL) {
+        errlogSevPrintf(errlogMajor, "Error in memory allocation of new_extra_envs from addReccasterEnvVars\n");
+        epicsMutexUnlock(self->lock);
+        return;
+    }
+    /* copy self->extra_envs into new_extra_envs with room for new envs */
+    for(i=0; i < self->num_extra_envs; i++) {
+        if((new_extra_envs[i] = strdup(self->extra_envs[i])) == NULL) {
+            errlogSevPrintf(errlogMinor, "strdup error from addReccasterEnvVars\n", new_extra_envs[j]);
+            ret = 1;
+            break;
+        }
+    }
+    int found_dup;
+    /* sanitize input - check for dups and empty args */
+    if(!ret) {
+        for(i=0; i < argc; i++) {
+            if(argv[i] == NULL) {
+                errlogSevPrintf(errlogMinor, "Arg is NULL for addReccasterEnvVars\n");
+                continue;
+            }
+            else if(argv[i][0] == '\0') {
+                errlogSevPrintf(errlogMinor, "Arg is empty for addReccasterEnvVars\n");
+                continue;
+            }
+            found_dup = 0;
+            /* check if dup in self->extra_envs. doesn't check if arg is in default_envs right now */
+            for(j = 0; j < num_new_extra_envs && new_extra_envs[j]; j++) {
+                if(strcmp(argv[i], new_extra_envs[j]) == 0) {
+                    found_dup = 1;
+                    errlogSevPrintf(errlogMinor, "Env var %s is already in extra_envs list\n", argv[i]);
+                    break;
+                }
+            }
+            if(found_dup) {
+                continue;
+            }
+            if((new_extra_envs[num_new_extra_envs] = strdup(argv[i])) == NULL) {
+                errlogSevPrintf(errlogMinor, "strdup error for copying %s to new_extra_envs[%d] from addReccasterEnvVars\n", new_extra_envs[num_new_extra_envs]);
+                ret = 1;
+                break;
+            }
+            /* this is a valid arg and we have added the new env var to our array, increment new_extra_envs count */
+            num_new_extra_envs++;
+        }
+    }
+    /* if we have no allocation issues and have at least one new env var that is valid, add to self->extra_envs */
+    if(!ret && num_new_extra_envs - self->num_extra_envs != 0) {
```

Comment:
Any particular reason you check the difference and not just num_new_extra_envs != self->num_extra_envs?

(or even num_new_extra_envs > self->num_extra_envs as a bit of a sanity check)

(this comment was added accidentally on the file, copied here)

Reply:
nope no reason besides not thinking right :)